File: Programmers/pro_12977_소수 만들기.py
isPrime = [1]*3001
def eratos(): #방법 1 공간을 써서 시간을 절약
    isPrime[0]=0
    isPrime[1]=0
    for i in range(2,len(isPrime)):
        if(isPrime[i]==1):
            j = 2
            while(i*j<len(isPrime)):
                isPrime[i*j]=0
                j+=1
                
def eratos2(num): #방법 2 시간을 써서 공간을 절약
    i=2
    while(i*i<=num):
        if(num%i==0):
            return False
        i+=1
    return True

def solution(nums):
    eratos()
    candidates = set()
    
    lenNum = len(nums)
    answer = 0
    for i in range(lenNum):
        for j in range(i+1,lenNum):
            for k in range(j+1,lenNum):
                # if(isPrime[(nums[i]+nums[j]+nums[k])]):
                    # answer+=1
                if(eratos2(nums[i]+nums[j]+nums[k])):
                    answer+=1
    
    return answer

File: Programmers/pro_12977_소수만들기.js
function isPrime(num) {
    for (let i = 2; i < Math.sqrt(num) + 1; i++) {
        if (parseInt(num % i) == 0) {
            return false
        }
    }
    return true
}

function solution(nums) {
    let answer = 0
    for (let i = 0; i < nums.length; i++)
        for (let j = i + 1; j < nums.length; j++)
            for (let k = j + 1; k < nums.length; k++)
                if (isPrime(nums[i] + nums[j] + nums[k]))
                    answer += 1
    return answer;
}